apis/Google.Devtools.AspNet/Google.Devtools.AspNet.Tests/CloudTrace/Utils.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Moq;
+using System.Collections.Generic;
+
+namespace Google.Devtools.AspNet.Tests
+{
+    public static class Utils
+    {
+        /// <summary>
+        /// Creates a rate limiter that will allow 1 QPS with the elapsed milliseconds.
+        /// </summary>
+        internal static RateLimiter GetRateLimiter(long elapsedMilliseconds)
+        {
+            Mock<ITimer> watch = new Mock<ITimer>();
+            watch.Setup(w => w.GetElapsedMilliseconds()).Returns(elapsedMilliseconds);
+            return new RateLimiter(1, watch.Object);
+        }
+
+        /// <summary>
+        /// Creates a rate limiter with the given QPS that returns the given elapsed milliseconds
+        /// in order.
+        /// </summary>
+        internal static RateLimiter GetRateLimiter(double qps, long[] elapsedMilliseconds)
```

Comment:
You can mark the last parameter as `params` so you don't need `new long[] {...}` at the call sites, but perhaps this is intentionally left out to avoid confusing the first argument with subsequent arguments.